### ISSUE TITLE
GPU Insufficient memory！！

### ISSUE THREAD

**Issue by pengzhiwang** [eriklindernoren/PyTorch-YOLOv3#40]
Hello
My GPU is 1080T. However, There has the problem of insufficient video memory when training. I think whether the subdivisions for the every batch  is need.??

**Comment by pengzhiwang** [comment]
Due to limited conditions， I can only use one GPU

**Comment by pengzhiwang** [comment]
Whether the code should be changed to the following form？

_________________________________________________________
batch_size      = int(hyperparams['batch'])
subdivisions    = int(hyperparams['subdivisions'])
sub_batch       = batch_size//subdivisions

.................

for epoch in range(opt.epochs):
    for batch_i, (_, imgs, targets) in enumerate(dataloader):
        imgs = Variable(imgs.type(Tensor))
        targets = Variable(targets.type(Tensor), requires_grad=False)

        optimizer.zero_grad()
        for i in range(subdivisions):
            img = imgs[i*sub_batch:(i+1)*sub_batch]
            target = targets[i*sub_batch:(i+1)*sub_batch]
            loss = model(img, target)
            loss.backward()
            print('[Epoch %d/%d, Batch %d/%d] [Losses: x %f, y %f, w %f, h %f, conf %f, cls %f, total %f, recall: %.5f]' %
                (epoch, opt.epochs, batch_i, len(dataloader),
                model.losses['x'], model.losses['y'], model.losses['w'],
                model.losses['h'], model.losses['conf'], model.losses['cls'],
                loss.item(), model.losses['recall']))

        optimizer.step()
        model.seen += imgs.size(0)

    if epoch % opt.checkpoint_interval == 0:
        model.save_weights('%s/%d.weights' % (opt.checkpoint_dir, epoch))

**Comment by mark-hoffmann** [comment]
You don't need to create sub_batches, just simply lower your batch size and you should be fine. It essentially does the same thing and that's what that parameter is for.
